tests/model_serving/model_server/upgrade/conftest.py

```
@@ -1430,3 +1402,608 @@ def _capture_and_save_llmd_baseline(
+    kueue_dsc_state_cm_name = "upgrade-kueue-dsc-state"
+
+    if pytestconfig.option.post_upgrade:
+        yield
+        _restore_kueue_dsc_state(
+            admin_client=admin_client,
+            dsc_resource=dsc_resource,
+            namespace=namespace,
+            kueue_dsc_state_cm_name=kueue_dsc_state_cm_name,
+        )
+    else:
+        from tests.model_serving.model_server.conftest import _is_kueue_operator_installed  # noqa: NIC001
```

Ruff (0.15.20)

[warning] Invalid rule code in `# noqa`: NIC001

Add non-Ruff rule codes to the `lint.external` configuration option

(RUF102)